File: 6-zigzag-conversion/zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int n) 
    {
        vector<vector<char>> vec(n);
        int k=0;
        while(k<s.size())
        {
            for(int i=0;i<n && k<s.size();i++)
                vec[i].push_back(s[k++]);
            for(int i=n-2;i>=1 && k<s.size();i--)
                vec[i].push_back(s[k++]);
        }
        string st="";
        for(int i=0;i<n;i++)
            for(char j:vec[i])
                st+=j;
        return st;

    }
};

File: 6-zigzag-conversion/zigzag-conversion.java
class Solution {
    public String convert(String s, int n) 
    {
        List<List<Character>> arr = new ArrayList<> ();
        for(int i=0;i<n;i++)
            arr.add( new ArrayList<>());

        int k=0;
        while(k<s.length())
        {
            for(int i=0;i<n && k<s.length();i++)
                arr.get(i).add(s.charAt(k++));
            for(int i=n-2;i>=1 && k<s.length();i--)
                arr.get(i).add(s.charAt(k++));
        }
        String asshoke="";
        for(int i=0;i<n;i++)
        {
            for(char ch:arr.get(i))
                asshoke+=ch;
        }
        return asshoke;

    }
}